### DISCUSSION TITLE
define relation between two walls with IfCOpenShell-Python

### DISCUSSION THREAD

**Discussion by NkStll** [IfcOpenShell/IfcOpenShell#3435] [Q&A]
Dear all,

is there currently a way with the python API to generate a relation between two walls?
(E.g. a horizontal wall and a vertical wall that is positioned on the surface of the horizontal wall)

In the IFC file this is represented with "IfcRelConnectsPathElements", if I understand it correctly.

Thanks for your help!
Best,
Nkstll

**Comment by Moult** [comment]
https://blenderbim.org/docs-python/autoapi/ifcopenshell/api/geometry/connect_path/index.html

The settings are:

```
        self.settings = {
            "relating_element": None,
            "related_element": None,
            "relating_connection": "NOTDEFINED",
            "related_connection": "NOTDEFINED",
            "description": None,
        }
```

Unfortunately this function is not documented but feel free to use it. https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/test/api/geometry/test_connect_path.py
